doc/contributing/writing-tests.md

```diff
@@ -12,6 +12,8 @@ with code `0` on success. A test will fail if:
 * It never exits. In this case, the test runner will terminate the test because
   it sets a maximum time limit.
 
+it is recommended to [use `node:test`][] for handling these requirements
```

Comment:
What's the motivation behind this recommendation? How does the test runner handle these two specific requirements better than a plain JS file without the overhead of having to use `test()` etc.?

For the first requirement, in virtually all cases, an unexpected error in a plain JS file will set `process.exitCode` correctly.

For the second requirement, the external test runner provides a reliable timeout mechanism, whereas the `timeout` mechanism that was added to `node:test` is unreliable.

Reply:
tests runners and specifically `node:test` have a few added values over running plain JS:
 - focus both on what fails and what does not. running plain code will stop immediately and will cause longer iterations to find all failures. when tests are broken down into small independent pieces with good names that usually provides very good feedback regarding what might be broken.
 - standard reporting: following the previous bullet, `node:test` reports an output that is readable by most CI's (/is converted to such an output), and provides a very specific output of what has failed, directly in the CI UI - instead of what file has failed
 - manage concurrency: assuming each file contains tests for more than a single behavior, a test runner can run more than one concurrent behaviors/tests, and make things faster.
 - readability - breaking down a file into small independent pieces makes it more readable IMHO

regarding the `timeout` reliability - I am not sure why you consider python's timeout being more reliable than node.js's timeouts, but this PR does not suggest replacing the external python test runner with `node:test`.

Reply:
this is obviously my personal experience with test runners, and you might have other experiences,
but I don't think that it is fair to claim that `node:test` has no value or that it is totally unreliable

Reply:
> but I don't think that it is fair to claim that `node:test` has no value or that it is totally unreliable

I am aware of the (potential) benefits of the built-in test runner, but that wasn't my question. The recommendation in this PR is worded as follows:

> it is recommended to use `node:test` for handling these requirements

This implies that the test runner has benefits that are specific to "these requirements", and my question remains

> How does the test runner handle these two specific requirements better than a plain JS file without the overhead of having to use `test()` etc.?

Reply:
I agree with @tniessen that we should understand what value having internally tests use the runner adds and I also think we should not make our internal test suite depdendent on it while it is still experimental and would also be concerned if it would make it harder to backport tests to previous versions.  If the value is super high it might override these concerns, but it would be good to factor these into any decision.